File: support_tools/swigwin-3.0.12/Examples/test-suite/php/pointer_reference_runme.php
<?php

require "tests.php";
require "pointer_reference.php";

$s  = pointer_reference::get();
check::equal($s->value, 10, "pointer_reference::get() failed");

$ss = new Struct(20);
pointer_reference::set($ss);
$i = Struct::instance();
check::equal($i->value, 20, "pointer_reference::set() failed");

check::equal(pointer_reference::overloading(1), 111, "overload test 1 failed");
check::equal(pointer_reference::overloading($ss), 222, "overload test 2 failed");

check::done();
?>

File: support_tools/swigwin-3.0.12/Examples/test-suite/csharp/pointer_reference_runme.cs
// This test tests all the methods in the C# collection wrapper

using System;
using pointer_referenceNamespace;

public class pointer_reference_runme {

  public static void Main() {
    Struct s  = pointer_reference.get();
    if (s.value != 10) throw new Exception("get test failed");

    Struct ss = new Struct(20);
    pointer_reference.set(ss);
    if (Struct.instance.value != 20) throw new Exception("set test failed");

    if (pointer_reference.overloading(1) != 111) throw new Exception("overload test 1 failed");
    if (pointer_reference.overloading(ss) != 222) throw new Exception("overload test 2 failed");
  }

}

